lib/room.rb

```diff
@@ -0,0 +1,49 @@
+require_relative 'reservation'
+
+module Hotel
+  class Room
+    attr_reader :room_id, :rez_list
+
+    def initialize(room_id)
+      @room_id = room_id
+      @rez_list = []
+
+      raise ArgumentError if room_id <= 0 || room_id > 20
+    end
+
+    def add_room_reservation(reservation_to_add)
+      @rez_list << reservation_to_add
+    end
+
+    def conflict?(input_range)
```

Comment:
You have mostly got really good method names, but `conflict?` is not quite descriptive enough here. Your asking the room if it conflicts, but that isn't how we'd ask that question conversationally. I'd opt for `is_booked?` or something similar, that communicates the _nature_ of the conflict.